### DISCUSSION TITLE
feat: Shopify Integration

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#8454] [Feature Requests]
- Ability to view Shopify orders and customer info in the chatwoot contact sidebar 
- Ability to perform actions like issue refund right from chatwoot dashboard

ref: https://discord.com/channels/647412545203994635/719548300868452362/972145068234375208

**Comment by substr2837** [comment]
Can I work on this issue?

**Comment by sojan-official** [comment]
@substr2837  Since this issue doesn't have detailed spec. Please share the approaches you considering before writing a significant amount of code so that there wouldn't be any confusion.

**Comment by substr2837** [comment]
Here are my approaches:
1) Add the integration page for Shopify. That page will be used to store the API key and API secret of Shopify
2) If the customer sends the message that contains the keyword "Shopify Order", the application will automatically reply that chat with a text box and ask the customer the order number
The order number will be stored on DB.
3) This order number will be used to receive order info from Shopify Admin API ([/admin/orders.json?number=](https://vfiles.myshopify.com/admin/orders.json?name=2275)number). This API response will also include customer and transactions info which the information will be stored in the DB.
4) This stored customer info will be displayed on the contact sidebar and the transaction data will be used when issuing refunds using this endpoint (admin/API/{{api_version}}/orders/{{order_id}}/refunds.json)

Please review and correct my approaches, if there is a wrong approach. Thanks

**Comment by sojan-official** [comment]
@substr2837 That's a good start. How about this approach? 

1. Add the integration page for Shopify. That page will be used to store the API key and API secret of Shopify  
2. Once the integration is added, a job is initiated that will sync all the customers in the Shopify store to Chatwoot as contacts ( matching them with email and phone number ) ( the shopify_id can be added to the additional attributes jsonb column ) ( Chatwoot will keep the Shopify customer list synced )
3. Whenever an agent talks to the contact -> if shopify_id is available, the sidebar will showcase all the orders associated with this contact / have provision to issue refunds, etc 

## Thoughts
- The agent has the full picture of the Customer right away at the Chat, So they can proactively talk about specific orders without requiring information from the Customer about order number 
- We can also leverage the marketing capabilities of Chatwoot like campaigns in future 
- We can also look into embedding the Chatwidget directly to the Shopify template in the future automatically 

## Notes

- After the initial sync we can rely on the webhooks to keep the customer info updated 
https://shopify.dev/api/admin-rest/2022-04/resources/webhook


let me know your thoughts.

**Comment by substr2837** [comment]
I think is its better approach @sojan-official . Huge thanks for reviewing my approaches and for sharing a really good thoughts. I will start working on this feature then.

**Comment by sojan-official** [comment]
@substr2837  no problem. Let us know if you need any help.

**Comment by tomryanx** [comment]
Is there a recommended stopgap solution to get the chatwoot widget showing on Shopify?

**Reply by sojan-official** [reply]
you can include the javascript directly

**Comment by sojan-official** [comment]
ref: ref: https://github.com/chatwoot/chatwoot/pull/11101
